toolchain/musl/common.mk

```diff
@@ -28,6 +28,8 @@ HOST_BUILD_DIR:=$(BUILD_DIR_TOOLCHAIN)/$(PKG_NAME)-$(PKG_VERSION)
 include $(INCLUDE_DIR)/host-build.mk
 include $(INCLUDE_DIR)/hardening.mk
 
+TARGET_CFLAGS := $(filter-out -O%,$(TARGET_CFLAGS))
```

Comment:
What happens if TARGET_CFLAGS matches the projects’s default settings?

Reply:
I'm not sure what you mean, it just strips out -O* (whatever is set by default)

https://git.musl-libc.org/cgit/musl/tree/configure#n407
I think that'll answer your question?